projects/gateway2/query/query.go

```diff
@@ -126,10 +127,20 @@ type GatewayQueries interface {
 
 	GetLocalObjRef(ctx context.Context, from From, localObjRef apiv1.LocalObjectReference) (client.Object, error)
 
-	// GetRoutesForGateway finds the top level xRoutes attached to the provided Gateway
-	GetRoutesForGateway(ctx context.Context, gw *apiv1.Gateway) (*RoutesForGwResult, error)
 	// GetRouteChain resolves backends and delegated routes for a the provided xRoute object
 	GetRouteChain(ctx context.Context, obj client.Object, hostnames []string, parentRef apiv1.ParentReference) *RouteInfo
+
+	// GetListenerSetsForGateway returns the list of listener sets mapped to the given gateway
+	// It consists of the list of allowed listener sets and denied listener sets (based on the value of allowedListeners)
+	GetListenerSetsForGateway(ctx context.Context, gw *apiv1.Gateway) (*ListenerSetsForGwResult, error)
```

Comment:
* GetListenerSetsForGateway seems like an internal detail of ConsolidateGateway
* GetRoutesForListenerSet seems like an internal detail of GetRoutesForConsolidatedGateway

we should probably hide the concept of listener sets from the rest of the codebase where possible

Reply:
Yeah makes sense